config/config.go

```diff
@@ -121,12 +134,35 @@ type ServerConfig struct {
 // APIConfig holds the REST API configuration
 type APIConfig struct {
 	Auth                ServiceAuthConfig `json:"-"`
+	CORS                CORSConfig        `json:"cors"`
 	MaxConns            int               `json:"maxConns"`
 	Port                string            `json:"port"`
 	EnableNotifications bool              `json:"enableNotifications"`
 	EnableOverrides     bool              `json:"enableOverrides"`
 }
 
+// CORSConfig holds the CORS middleware configuration
+type CORSConfig struct {
+	AllowedOrigins     []string `json:"allowedOrigins"`
+	AllowedMethods     []string `json:"allowedMethods"`
+	AllowedHeaders     []string `json:"allowedHeaders"`
+	ExposedHeaders     []string `json:"exposedHeaders"`
+	AllowedCredentials bool     `json:"allowedCredentials"`
+	MaxAge             int      `json:"maxAge"`
+}
+
+// ToCorsOptions converts CORSConfig to cors middleware options
+func (c *CORSConfig) ToCorsOptions() cors.Options {
```

Comment:
Nit, but I'd prefer we didn't add this here as it feels like we're leaking a dependency. I think having a method in `api.go` would be a cleaner abstraction.
```
func CreateCorsHandler(c conf.CORSConfig) func(next http.Handler) http.Handler {
  options := cors.Options{
		AllowedOrigins:   c.AllowedOrigins,
		AllowedMethods:   c.AllowedMethods,
		AllowedHeaders:   c.AllowedHeaders,
		ExposedHeaders:   c.ExposedHeaders,
		AllowCredentials: c.AllowedCredentials,
		MaxAge:           c.MaxAge,
	}

  return cors.Handler(options)
}
```